Added callback support and as async (requires change to libsignal-protocol-dotnet for it)

mitchcapper opened the pull request:
If we didn't want to do async/Task support we could avoid changing the protocol library, (and the library itself has to use .wait as it has locks). This is a bit more future proof / forward planning per the direction it seems you want to be going.   Suggest using ```?w=1``` when reviewing the changes to see the content items.

https://github.com/signal-csharp/libsignal-protocol-dotnet/pull/5

Closes #18

Trolldemorted commented:
Do we really need the asynchronity? I think it doesn't make that much sense that deep in the stack, especially with the locks around.

mitchcapper commented:
Updated Add GetStrippedMessage that takes a SessionRecord instead and use it so callback support on new sessions works.  I described this in #18 but didn't submit up the change.

mitchcapper committed:
Added callback support and as async (requires change to libsignal-protocol-dotnet for it)
Add GetStrippedMessage that takes a SessionVersion instead and use it so callback support on new sessions works
